### PR TITLE
Added 3 function to sdk,  to call embedding, ref-embedding and process

### PR DESCRIPTION
**Pull request by kundan1301**

### CODE REVIEW

File: markov_ai/components/transformation.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import logging
+from typing import Any, Dict, Optional
+from ..service import post_json_data
+from ..constants import MARKOV_AI_BASE_URL, MARKOV_AI_EMBEDDINGS_ENDPOINT,MARKOV_AI_REF_EMBEDDING_ENDPOINT
+
+
+class Transformation:
```
Comment (AnantChandra) [inline markov_ai/components/transformation.py]: Minor: Can we rename this to `Transformer`

File: markov_ai/components/pre_processor.py [hunk 2]
```diff
@@ -49,3 +49,19 @@ def _handle_response(response: Optional[Any], operation: str) -> Optional[Dict[s
             return response.json()
         logging.error(f"{operation} failed: {response.text if response else 'No response'}")
         return None
+
+    def process(self, markov_s3_key: str, **kwargs: Any) -> Optional[Dict[str, Any]]:
```
Comment (AnantChandra) [inline markov_ai/components/pre_processor.py:38]: Can we please put this method above the helper methods

File: markov_ai/components/transformation.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    def __init__(self, **kwargs: Any) -> None:
+        self.__dict__.update(kwargs)
+        self.api_key = kwargs.get('api_key')
+
```
Comment (AnantChandra) [inline markov_ai/components/transformation.py]: Let's put `get_headers` & `handle_response` at the bottom of the file

File: markov_ai/components/transformation.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import logging
+from typing import Any, Dict, Optional
+from ..service import post_json_data
+from ..constants import MARKOV_AI_BASE_URL, MARKOV_AI_EMBEDDINGS_ENDPOINT,MARKOV_AI_REF_EMBEDDING_ENDPOINT
+
+
```
Comment (AnantChandra) [inline markov_ai/components/transformation.py]: Please export the class here:

- https://github.com/markov-ai-xyz/markov-ai-python-sdk/blob/main/markov_ai/components/__init__.py

- https://github.com/markov-ai-xyz/markov-ai-python-sdk/blob/main/markov_ai/__init__.py

File: markov_ai/components/transformation.py (hunk 3 above)
Comment (kundan1301) [reply markov_ai/components/transformation.py]: sure

File: markov_ai/components/transformation.py (hunk 4 above)
Comment (kundan1301) [reply markov_ai/components/transformation.py]: done

**Review by AnantChandra — APPROVED** [review]
LGTM. Have you tested this code via a Jupyter notebook or such?

**Pull request merged**